# scRNAseq-analysis-notes
Selection of papers, tools, tutorials and notes for scRNAseq analyses in R from Haoyu Zhang and Niklas Lang during their MSc project @UniversityOfEdinburgh

## Papers and Tools

* [Integrated analyses of single-cell atlases: cell type-specific expression of mediators of SARS-CoV-2 viral entry ](https://www.biorxiv.org/content/10.1101/2020.04.19.049254v2)
* [SARS-CoV-2 entry factors in single-cell RNA-sequencing data from multiple tissues from healthy human donors](https://www.nature.com/articles/s41591-020-0868-6)

## Tutorials

* [Seurat (Overview)](https://satijalab.org/seurat/vignettes.html)

## Notes and Thoughts on ...

### QC

### Data Normalization

### Data Correction

### Feature Selection and Dimensionality Reduction

### Clustering

* [Avoiding common pitfalls when clustering biological data](https://stke.sciencemag.org/content/9/432/re6)
* [How to cluster in High Dimensions](https://towardsdatascience.com/how-to-cluster-in-high-dimensions-4ef693bacc6)
* [Challenges in unsupervised clustering of single-cell RNA-seq data](https://www.nature.com/articles/s41576-018-0088-9)
* [Comparison of Clustering Tools in R for Medium-Sized 10x Genomics Single-Cell RNA-sequencing Data](https://f1000research.com/articles/7-1297)
* [A systematic performance evaluation of clustering methods for single-cell RNA-seq data](https://f1000research.com/articles/7-1141)

### Cluster Annotation

### Visualization

* [UMAP (Paper)](https://www.nature.com/articles/nbt.4314)

### Dataset Integration

* [Seurat multi-CCA (Paper)](https://www.nature.com/articles/nbt.4096)
* [MNN correct (Paper)](https://www.nature.com/articles/nbt.4091)
* [Harmony (Paper)](https://www.nature.com/articles/s41592-019-0619-0)
* [LIGER (Paper)](https://www.cell.com/cell/fulltext/S0092-8674%2819%2930504-5)
* [LIGER (GitHub)](https://github.com/MacoskoLab/liger)
* [conos (Paper)](https://www.nature.com/articles/s41592-019-0466-z)
* [conos (GitHub)](https://github.com/hms-dbmi/conos)

### Challenges

* [Computational and analytical challenges in single-cell transcriptomics](https://www.nature.com/articles/nrg3833)
